plugin/trino-resource-group-managers/src/main/java/io/trino/plugin/resourcegroups/StaticSelector.java

```diff
@@ -81,26 +89,24 @@ public Optional<SelectionContext<ResourceGroupIdTemplate>> match(SelectionCriter
     {
         Map<String, String> variables = new HashMap<>();
 
-        if (userRegex.isPresent()) {
-            Matcher userMatcher = userRegex.get().matcher(criteria.getUser());
-            if (!userMatcher.matches()) {
-                return Optional.empty();
-            }
-
-            addVariableValues(userRegex.get(), criteria.getUser(), variables);
+        if (!addVariablesForRegexIfMatching(userRegex, criteria.getUser(), variables)) {
+            return Optional.empty();
         }
 
         if (userGroupRegex.isPresent() && criteria.getUserGroups().stream().noneMatch(group -> userGroupRegex.get().matcher(group).matches())) {
             return Optional.empty();
         }
 
-        if (sourceRegex.isPresent()) {
-            String source = criteria.getSource().orElse("");
-            if (!sourceRegex.get().matcher(source).matches()) {
-                return Optional.empty();
-            }
+        if (!addVariablesForRegexIfMatching(originalUserRegex, criteria.getOriginalUser(), variables)) {
+            return Optional.empty();
+        }
 
-            addVariableValues(sourceRegex.get(), source, variables);
+        if (!addVariablesForRegexIfMatching(authenticatedUserRegex, criteria.getAuthenticatedUser().orElse(""), variables)) {
+            return Optional.empty();
+        }
+
+        if (!addVariablesForRegexIfMatching(sourceRegex, criteria.getSource().orElse(""), variables)) {
```

Comment:
do we want to match against empty string? wrap with ifPresent()

Reply:
Moving this discussion above: https://github.com/trinodb/trino/pull/24662#discussion_r1958286302